src/frontend/apps/desk/src/features/teams/member-management/components/MemberGrid.tsx

```diff
@@ -43,30 +48,64 @@ const defaultOrderingMapping: Record<string, string> = {
  */
 function formatSortModel(
   sortModel: SortModelItem,
-  mapping = defaultOrderingMapping,
-) {
+  mapping: Record<string, string> = defaultOrderingMapping,
+): string {
   const { field, sort } = sortModel;
   const orderingField = mapping[field] || field;
   return sort === 'desc' ? `-${orderingField}` : orderingField;
 }
 
+const MS_DEBOUNCE = 500;
+export function useDebounce(value: string, delay: number = MS_DEBOUNCE) {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+
+  return debouncedValue;
+}
+
 export const MemberGrid = ({ team, currentRole }: MemberGridProps) => {
   const [isModalMemberOpen, setIsModalMemberOpen] = useState(false);
   const { t } = useTranslation();
   const { colorsTokens } = useCunninghamTheme();
   const pagination = usePagination({
     pageSize: PAGE_SIZE,
   });
+
+  const searchMemberValidationSchema = z.object({
+    query: z.string().optional(),
+  });
+
+  const methods = useForm<{ query: string }>({
+    delayError: 0,
+    defaultValues: {
+      query: '',
+    },
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+    resolver: zodResolver(searchMemberValidationSchema),
+  });
+
```

Comment:
I don't think we should use `useForm` here, it just make the code more complex when a simple `onChange` on the input will be just enough.